shell/common/shell_unittests.cc

```diff
@@ -2014,5 +2014,77 @@ TEST_F(ShellTest, OnServiceProtocolEstimateRasterCacheMemoryWorks) {
   DestroyShell(std::move(shell));
 }
 
+TEST_F(ShellTest, DiscardLayerTreeOnResize) {
+  auto settings = CreateSettingsForFixture();
+
+  std::shared_ptr<ShellTestExternalViewEmbedder> external_view_embedder;
+
+  SkISize wrong_size = SkISize::Make(400, 100);
+  SkISize expected_size = SkISize::Make(400, 200);
+
+  fml::AutoResetWaitableEvent end_test_latch;
+
+  auto post_frame_callback = [&](bool, fml::RefPtr<fml::RasterThreadMerger>) {
+    // First submitted frame must be expected size even though first rendered
+    // tree is wrong size
+    if (external_view_embedder->GetSubmittedFrameCount() == 1) {
+      ASSERT_EQ(external_view_embedder->GetLastSubmittedFrameSize(),
+                expected_size);
+      end_test_latch.Signal();
+    }
+  };
+
+  external_view_embedder = std::make_shared<ShellTestExternalViewEmbedder>(
+      std::move(post_frame_callback), PostPrerollResult::kSuccess, true);
+
+  std::unique_ptr<Shell> shell = CreateShell(
+      settings, GetTaskRunnersForFixture(), false, external_view_embedder);
+
+  // Create the surface needed by rasterizer
+  PlatformViewNotifyCreated(shell.get());
+
+  fml::TaskRunner::RunNowOrPostTask(
+      shell->GetTaskRunners().GetPlatformTaskRunner(),
+      [&shell, &expected_size]() {
+        shell->GetPlatformView()->SetViewportMetrics(
+            {1.0, double(expected_size.width()),
+             double(expected_size.height())});
+      });
+
+  auto configuration = RunConfiguration::InferFromSettings(settings);
+  configuration.SetEntrypoint("emptyMain");
+
+  RunEngine(shell.get(), std::move(configuration));
+
+  fml::WeakPtr<RuntimeDelegate> runtime_delegate = shell->GetEngine();
+
+  auto render_tree = [&](const SkISize& size) {
+    shell->GetTaskRunners().GetUITaskRunner()->PostTask(
+        [runtime_delegate, size]() {
+          auto layer_tree =
+              std::make_unique<LayerTree>(size, static_cast<float>(1));
```

Comment:
nit: this can be 1.0f.

Reply:
Indeed. Not sure why it was written like this.